### ISSUE TITLE
Please add instructions on how to run it in the README

### ISSUE THREAD

**Issue by Kapsyloffer** [klautcomputing/hive-leptos#106]
I'd really like to mess around with it, thank you.

**Comment by klautcomputing** [comment]
Hi @Kapsyloffer the setup is a bit more complicated as you require a setup postgresql db, rust nightly, diesel, leptos watch ... 
I will make write something up, but depending on your OS, it might only get you so far before you need to experiment a bit on your own. 
What OS are you on?

**Comment by klautcomputing** [comment]
Let's turn this conversation into a readme afterwards, I think that makes it a bit easier for both of us :)

You will need:
- clone this repo
- install `rustup` https://rustup.rs/ or through your package manager
- then run `rustup toolchain install nightly` to get nightly, then most likely `rustup default nightly` to also use it permanently
- install https://github.com/leptos-rs/cargo-leptos
- make sure you have postgres installed (and the dev packages as well)
- then install diesel via `cargo install diesel_cli --no-default-features --features postgres`
- create a postgres database called "hive-local" that the user "hive-dev" has access to, take a look at the `.env` file 
- go into `/db` and run `diesel migration run`
- not needed but cool: install https://github.com/rui314/mold for better linking times
- install dart-sass and openssl dependencies
- now run: `cargo leptos watch` or `mold -run cargo leptos watch` from the main directory
- report back and we fix wherever you got stuck, which I expect to happen because I might have forgotten something... 

@IongIer did I forget anything?

**Comment by Kapsyloffer** [comment]
Is installing leptos-rs via github necessary when it's already in the Cargo.toml?

**Comment by Kapsyloffer** [comment]
Also maybe add this link https://www.postgresql.org/download/ in the README for ease of use :P 

Also by all dev packages, do you mean everything from the stack builder?

![image](https://github.com/klautcomputing/hive-leptos/assets/32063345/56938064-7f42-470c-b91a-3899959a47c0)

**Comment by IongIer** [comment]
Cargo-leptos is a build tool for leptos and it needs to be installed separately. It also handles the dart-sass part and a few other needed things so those don't need to be installed separately afaik.

```
create a postgres database called "hive-local" that the user "hive-dev" has access to, take a look at the .env file
go into /db and run diesel migration run
```
These two steps might be incomplete, in my case the migration failed first time and I needed to alter "hive-dev" and grant it some additional things. If you get to this point and it doesn't work I'll do my best to help you, but don't remember atm exactly what I needed to do, postgres logs helped me figure out the problem. I've only had to get it working on arch and in WSL2 ubuntu on windows, no experience with straight windows for me yet.

**Comment by klautcomputing** [comment]
> Is installing leptos-rs via github necessary when it's already in the Cargo.toml?

Yes, you will still need to do `cargo install --locked cargo-leptos` For the reason @IongIer mentioned.

**Comment by klautcomputing** [comment]
> Also maybe add this link https://www.postgresql.org/download/ in the README for ease of use :P

I don't want to rude or anything, but I think where you get your postgres from is very much up to you and by that I mean your operating system. :slightly_smiling_face: For me, I just mention it in my nix files, others will use Arch or Debian or whatever, so it will be hard to cover all of them.
 
> Also by all dev packages, do you mean everything from the stack builder?

No, it's just that operating systems often come with separate packages for the executable (which we need to run postgresql) and libraries (that we need to build against). With "dev" I meant "libs", so that you should make sure you have the libraries as well so that cargo can install diesel with postgres as a backend.
I saw that Manjaro is Arch based so you'd need something like this https://archlinux.org/packages/extra/x86_64/postgresql-libs/ and not just https://archlinux.org/packages/extra/x86_64/postgresql/. But again I don't know whether those are the right packages, I never used Arch :)
For windows, tbh I don't know. But @IongIer got it to run on WSL once, iirc.

Feel free to join https://discord.gg/sHbtKAvDvS and find us there and we can help you more synchronously :slightly_smiling_face:

**Comment by klautcomputing** [comment]
The readme now has some information on how to setup the project.
